File: sdks/python/aisandbox.py
import requests
import json
from typing import List, Dict, Any, Optional, Generator

class AISandboxClient:
    def __init__(self, base_url: str = "http://localhost:8000", api_key: str = None):
        self.base_url = base_url
        self.api_key = api_key
        self.chat = Chat(self)

class Chat:
    def __init__(self, client: AISandboxClient):
        self.client = client

    def completions(self, provider: str, model: str, messages: List[Dict[str, str]], stream: bool = False, **kwargs) -> Any:
        url = f"{self.client.base_url}/api/v1/chat/completions"
        headers = {
            "Content-Type": "application/json",
            "x-api-key": self.client.api_key
        }
        payload = {
            "provider": provider,
            "model": model,
            "messages": messages,
            "stream": stream,
            **kwargs
        }
        
        response = requests.post(url, json=payload, headers=headers, stream=stream)
        if response.status_code != 200:
            raise Exception(f"AI Sandbox Error: {response.text}")
        
        if stream:
            def gen():
                for line in response.iter_lines():
                    if line:
                        line = line.decode('utf-8')
                        if line.startswith("data: "):
                            data = json.loads(line[6:])
                            yield data
            return gen()
        
        return response.json()

File: sdks/javascript/aisandbox.js
class AISandboxClient {
    constructor(config = { baseUrl: "https://nexus-backend-ffn4.onrender.com", apiKey: "" }) {
        this.baseUrl = config.baseUrl || "https://nexus-backend-ffn4.onrender.com";
        this.apiKey = config.apiKey || "";
        this.chat = new Chat(this);
    }
}

class Chat {
    constructor(client) {
        this.client = client;
    }

    async completions(provider, model, messages, options = {}) {
        const stream = options.stream || false;
        const url = `${this.client.baseUrl}/api/v1/chat/completions`;
        const headers = {
            "Content-Type": "application/json",
            "x-api-key": this.client.apiKey
        };
        const body = JSON.stringify({
            provider: provider,
            model: model,
            messages: messages,
            ...options
        });

        const response = await fetch(url, {
            method: 'POST',
            headers,
            body
        });

        if (!response.ok) {
            const errText = await response.text();
            throw new Error(`AI Sandbox Error: ${errText}`);
        }

        if (stream) {
            const reader = response.body.getReader();
            const decoder = new TextDecoder();
            return {
                [Symbol.asyncIterator]: async function* () {
                    let buffer = "";
                    while (true) {
                        const { done, value } = await reader.read();
                        if (done) break;
                        buffer += decoder.decode(value, { stream: true });
                        const lines = buffer.split("\n");
                        buffer = lines.pop();
                        for (const line of lines) {
                            if (line.startsWith("data: ")) {
                                try {
                                    yield JSON.parse(line.slice(6));
                                } catch (e) {
                                    console.error("Failed to parse SSE line", line);
                                }
                            }
                        }
                    }
                }
            };
        }

        return await response.json();
    }
}

if (typeof module !== 'undefined' && module.exports) {
    module.exports = AISandboxClient;
} else {
    window.AISandboxClient = AISandboxClient;
}
